src/bartiq/analysis/rewriters/utils.py

```diff
@@ -31,22 +36,41 @@ class Comparators(str, Enum):
     LESS_THAN = "<"
 
 
-@dataclass
-class Assumption:
-    """A simple class for storing information about symbol assumptions."""
+class Instruction:
+    """Base class for all rewriter-transforming instructions."""
+
+
+@dataclass(frozen=True)
+class Initial(Instruction):
```

Comment:
[question] Why these aren't just enums?
Ok, I see later that we have some `__post_init__`, that's probably the reason?